### DISCUSSION TITLE
Option to disable diagnostic count limit

### DISCUSSION THREAD

**Discussion by MikeRomaa** [biomejs/biome#3060] [Ideas]
Hello! I'm currently working on an integration for biome with a custom continuous integration pipeline and I'd like to get *all* of the diagnostics on the entire repository in the report.

However, beyond just setting `--max-diagnostics` to something ridiculous like `1000`, I don't see a way of disabling the limit. Is there a way of achieving this? If not, would there be interest in such a option being added to Biome? Would love to implement it myself if so!

**Comment by GRA0007** [comment]
@MikeRomaa I think the `biome ci` command, which is designed for running in continuous integration does not implement a limit.

Related issue: when running with the `--reporter=summary` option, the max diagnostics limit should be disabled by default.

**Reply by ematipico** [reply]
Why? And what about other reporters? Don't you think this would involve too much magic?

**Reply by GRA0007** [reply]
Well I did assume that the limit was in place to avoid filling up your terminal history, but of course that's not an issue with the summary reporter. Additionally it could be confusing when running it if you don't notice the warning that it has only printed the first 10 errors.

I personally can't think of any reason to limit the amount on the summary, and it's not a good solution to provide a number like `--max-diagnostics=1000`; what if this is being used in a CI command and a project has over 1000 issues?

I agree that it could be confusing having different behaviour with different reporters, but I'd argue that the usability of the command suffers because of it. At the very least, the `--max-diagnostics` option should support a `none` or similar option.

**Reply by MikeRomaa** [reply]
For reasons I don't really want to get into, my tool also needs to have the ability to fix the files. `biome ci` doesn't allow this. Fundamentally, the two serve different purposes and I think such an option would make sense to have for `biome check`.

**Comment by ematipico** [comment]
How will this option look like?

**Reply by MikeRomaa** [reply]
Could be a separate flag like `--no-max-diagnostics` or a special value like `--max-diagnostics=none` as @GRA0007 suggested. I'm sure there's other options, but that's what I can think of.

**Comment by ematipico** [comment]
Closing, as being implemented in https://github.com/biomejs/biome/pull/3461
